code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/service/impl/ServiceDiscoveryServiceImpl.java

```diff
@@ -750,6 +760,16 @@ protected String calculateServiceHealthState(Service service) {
             int startedOnce = 0;
             List<String> runningStates = Arrays.asList(InstanceConstants.STATE_RUNNING);
             for (Instance instance : serviceInstances) {
+                if (isStartOnce(instance)) {
+                    startedOnce++;
+                }
+                if(isGlobal) {
+                    String hostId = DataAccessor.fieldString(instance, InstanceConstants.FIELD_HOST_ID);
+                    if(!activeHosts.contains(hostId)) {
```

Comment:
why are we doing it this way? It should be the other way around, if the activeHosts.contains, then increment healthy count

Reply:
https://github.com/kinarashah/cattle/blob/2672e4f9a64f4d4f856fad6904c2f2d781302278/code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/service/impl/ServiceDiscoveryServiceImpl.java#L791